src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
+    private RoleDao roleDao;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
+    }
+
+    @Test
+    void getRoleByName_validRoleName_ok() {
+        Role expected = new Role(Role.RoleName.USER);
+        roleDao.save(expected);
+        String roleUser = "USER";
+        Optional<Role> actual = roleDao.getRoleByName(roleUser);
+        Assertions.assertTrue(actual.isPresent());
+        Assertions.assertEquals(
+                expected.getRoleName(),
+                actual.get().getRoleName(),
+                "Method should return optional containing role "
+                        + expected.getRoleName()
+                        + " for roleName "
+                        + roleUser);
+    }
+
+    @Test
+    void getRoleByName_validRoleNameMissingInDb_ok() {
+        String roleUser = "USER";
```

Comment:
I'd use another role name here not to interfere with tests where we saving "USER" role and it may be present in db